### DISCUSSION TITLE
"export 'connect' (imported as 'mqtt') was not found in 'mqtt/dist/mqtt.min'

### DISCUSSION THREAD

**Discussion by jianjun-Web** [mqttjs/MQTT.js#1799] [Q&A] [answered]
Why is this a problem?
![image](https://github.com/mqttjs/MQTT.js/assets/70632135/291b44d0-fe0d-4ff9-9a42-dfa2ddc82ae5)
if use  var mqtt = require("mqtt/dist/mqtt.min.js")，is catch error
![image](https://github.com/mqttjs/MQTT.js/assets/70632135/e3b53173-d598-4abf-bb26-4dd71f14ef09)

**Comment by jianjun-Web** [comment]
Never mind the address， if use import mqtt from "mqtt";will have error
![image](https://github.com/mqttjs/MQTT.js/assets/70632135/fde50df5-0d9f-4549-a086-b1ce4e7b690b)
How to solve these problems

**Reply by robertsLando** [reply]
@jianjun-Web mqtt js version? If not latest please upgrade to latest and retry

**Reply by jianjun-Web** [reply]
is v4.3.7,node is v12，I tried the latest version and the error was the same

**Reply by robertsLando** [reply, marked as answer]
> is v4.3.7,node is v12，I tried the latest version and the error was the same

That's impossible because I'm using the latest version in browser in many projects and it's working.

Just install latest 5.3.6 and use this:

```ts
import mqtt from "mqtt"
```

Also consider using NodeJS 16/18 at least as NodeJS 12 has reached and of life on April 2022.
